Source/DafnyCore/Verifier/BoogieGenerator.Decreases.cs

```diff
@@ -106,28 +102,36 @@ public partial class BoogieGenerator {
   /// If builder is non-null, then the check '0 ATMOST decr' is generated to builder.
   /// Requires all types in types0 and types1 to be non-proxy non-synonym types (that is, callers should invoke NormalizeExpand)
   /// </summary>
-  Bpl.Expr DecreasesCheck(List<IToken> toks, List<Type> types0, List<Type> types1, List<Bpl.Expr> ee0, List<Bpl.Expr> ee1,
+  Bpl.Expr DecreasesCheck(List<IToken> toks, List<VarDeclStmt> prevGhostLocals,
+                          List<Expression> dafny0, List<Expression> dafny1, List<Bpl.Expr> ee0, List<Bpl.Expr> ee1,
                           BoogieStmtListBuilder builder, string suffixMsg, bool allowNoChange, bool includeLowerBound) {
     Contract.Requires(cce.NonNullElements(toks));
-    Contract.Requires(cce.NonNullElements(types0));
-    Contract.Requires(cce.NonNullElements(types1));
+    Contract.Requires(cce.NonNullElements(dafny0));
+    Contract.Requires(cce.NonNullElements(dafny1));
     Contract.Requires(cce.NonNullElements(ee0));
     Contract.Requires(cce.NonNullElements(ee1));
     Contract.Requires(predef != null);
-    Contract.Requires(types0.Count == types1.Count && types0.Count == ee0.Count && ee0.Count == ee1.Count);
+    Contract.Requires(dafny0.Count == dafny1.Count && dafny0.Count == ee0.Count && ee0.Count == ee1.Count);
     Contract.Requires(builder == null || suffixMsg != null);
     Contract.Ensures(Contract.Result<Bpl.Expr>() != null);
 
-    int N = types0.Count;
+    int N = dafny0.Count;
 
+    // TODO: NormalizeExpandKeepConstraints below?
```

Comment:
It looks like this TODO may be resolved?

Reply:
Indeed it is.